src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmTestCommand.cs

```diff
@@ -18,16 +22,25 @@ internal class WasmTestCommand : TestCommand
         protected override TestCommandArguments TestArguments => _arguments;
         protected override string CommandUsage { get; } = "wasm test [OPTIONS]";
         protected override string CommandDescription { get; } = CommandHelp;
+        public IEnumerable<string> PassThroughArgs => PassThroughArguments;
 
-        protected override Task<ExitCode> InvokeInternal(ILogger logger)
+        protected override async Task<ExitCode> InvokeInternal(ILogger logger)
         {
-            logger.LogDebug($"Wasm Test command called");
+            var processManager = new MacOSProcessManager();
+
+            // Expected syntax
+            // xharness wasm test --engine=v8 --engine-arg=--expose_wasm --engine-arg=--some-other-thing --js-file=runtime.js -- --enable-gc --run WasmTestRunner.dll System.Buffers.Tests.dll
+           
+            var result = await processManager.ExecuteCommandAsync(
+                _arguments.Engine.ToString(),
```

Comment:
Do you want to do `.ToLower()` here maybe? Or the binaries are called exactly like the enum values?

Reply:
The engine binaries invocation doesn't depend on a case, i.g. both `v8 ...` and `V8 ...` should work.

Reply:
Oh, ok. My other concern was that someone might change the enums, thinking it's a safe operation, but it would break this silent contract. Maybe update the docs of the enum at least to not that this silent contract exists?

Reply:
updated the doc